instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex

```diff
@@ -1,15 +1,41 @@
 defmodule OpentelemetryPhoenix do
+  @options_schema NimbleOptions.new!(
+                    endpoint_prefix: [
+                      type: {:list, :atom},
+                      default: [:phoenix, :endpoint],
+                      doc: "The endpoint prefix in your endpoint."
+                    ],
+                    adapter: [
+                      type: {:in, [:cowboy2, nil]},
+                      default: nil,
+                      doc: "The phoenix server adapter being used.",
+                      type_doc: ":atom"
+                    ]
+                  )
+
   @moduledoc """
   OpentelemetryPhoenix uses [telemetry](https://hexdocs.pm/telemetry/) handlers to create `OpenTelemetry` spans.
 
   Current events which are supported include endpoint start/stop, router start/stop,
   and router exceptions.
 
+  ### Supported options
+  #{NimbleOptions.docs(@options_schema)}
+
+  If you are using PlugCowboy as your adapter you can add `:opentelemetry_cowboy` to your project
+  and pass the `:adapter` option when calling setup. Setting this option will prevent a new
+  span from being started and the existing cowboy span to be continued. This is the recommended
+  setup for measuring accurate latencies.
+
+  `Bandit.PhoenixAdapter` is not currently supported.
+
   ## Usage
 
   In your application start:
 
       def start(_type, _args) do
+        :opentelemetry_cowboy.setup()
+        OpentelemetryPhoenix.setup(adapter: :cowboy2)
```

Comment:
Is there a way to detect, or just assume, this? Aren't the vast majority of uses cowboy2, why not make it that it is cowboy2 unless you supply an adapter option?

Reply:
We'd have to get a config added to specify the endpoint and then parse through the config to find which adapter is in use for it. We can't assume when there are multiple endpoints running because we're setting up the handlers based on the telemetry prefix. The other option is we have to parse through the conn on every request to try to find it which seems a little a 

No matter what, the user is going to have to install server instrumentation for the adapter they're using. I don't have qualms with making the user specify which one to use. It's also likely Bandit will become the goto within a year or two. Its performance is substantially better and has support in Phoenix 1.7. If it holds up, I could see it becoming the default in a future release.

We could make it the default, but if they don't add and setup cowboy instrumentation then it just won't work at all. Felt like a breaking change if we went that route. Or this lib needs to package up multiple adapters and install them whether they use them or not. I mean, Ecto requires you to say which adapter you're using through config. With Ecto repos, you're adding specific repos to the supervision tree to start them. Phoenix does the same with JSON library tied to an endpoint, but the endpoint again is added to the supervision tree. So at some place there's config happening and they're added to the start or supervision tree.

I dunno.